src/CommunityToolkit/Xamarin.CommunityToolkit/Views/HexLayout/HexLayout.shared.cs

```diff
@@ -0,0 +1,266 @@
+                BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) => ((HexLayout)bindable).InvalidateMeasure());
+
+        public int ColumnCount
+        {
+            get => (int)GetValue(ColumnCountProperty);
+            set => SetValue(ColumnCountProperty, value);
+        }
+
+        public static readonly BindableProperty RowCountProperty =
+            BindableProperty.Create(nameof(RowCount), typeof(int), typeof(HexLayout), 1,
+                BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) => ((HexLayout)bindable).InvalidateMeasure());
+
+        public int RowCount
+        {
+            get => (int)GetValue(RowCountProperty);
+            set => SetValue(RowCountProperty, value);
+        }
+
+        public static readonly BindableProperty ColumnProperty =
+            BindableProperty.Create(nameof(Column), typeof(int), typeof(HexLayout), 1,
+                BindingMode.TwoWay, null);
+
+        public int Column
+        {
+            get => (int)GetValue(ColumnProperty);
+            set => SetValue(ColumnProperty, value);
+        }
+
+        int GetColumn(VisualElement e) => Math.Min((int)e.GetValue(ColumnProperty), ColumnCount - 1);
+
+        public static readonly BindableProperty RowProperty =
+            BindableProperty.Create(nameof(Row), typeof(int), typeof(HexLayout), 1,
+                BindingMode.TwoWay, null);
+
+        public int Row
+        {
+            get => (int)GetValue(RowProperty);
+            set => SetValue(RowProperty, value);
+        }
+
+        int GetRow(VisualElement e) => Math.Min((int)e.GetValue(RowProperty), RowCount - 1);
+
+        protected override SizeRequest OnMeasure(double widthConstraint, double heightConstraint)
+        {
+            var w = widthConstraint;
+            var h = heightConstraint;
+
+            if (double.IsInfinity(w) || double.IsInfinity(h))
+            {
+                h = 0;
+                w = 0;
+
+                foreach (var child in Children)
+                {
+                    var childSize = child.Measure(widthConstraint, heightConstraint);
+
+                    if (childSize.Request.Height > h)
+                        h = childSize.Request.Height;
+
+                    if (childSize.Request.Width > w)
+                        w = childSize.Request.Width;
+                }
+
+                if (Orientation == HexOrientation.Horizontal)
+                    return new SizeRequest(new Size(w * ((ColumnCount * 3) + 1) / 4, h * ((RowCount * 2) + 1) / 2));
+
+                return new SizeRequest(new Size(w * ((ColumnCount * 2) + 1) / 2, h * ((RowCount * 3) + 1) / 4));
+            }
+
+            return Measure(widthConstraint, heightConstraint);
+        }
+
+        protected override void LayoutChildren(double x, double y, double width, double height)
+        {
+            HasShift(out var first, out var last);
```

Comment:
We aren't using the `last` value.
```suggestion
            HasShift(out var first, out _);
```